engines/dfc_provider/app/services/affiliate_sales_data_builder.rb

```diff
@@ -0,0 +1,171 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ClassLength
+class AffiliateSalesDataBuilder < DfcBuilder
+  class << self
+    def build_person(user)
+      DataFoodConsortium::Connector::Person.new(
+        urls.affiliate_sales_data_person_url(user.id),
+        logo: nil,
+        firstName: nil,
+        lastName: nil,
+        affiliatedOrganizations: user_enterprises(user.enterprises)
+      )
+    end
+
+    def user_enterprises(enterprises)
+      enterprises.map { |enterprise| build_enterprise(enterprise.id) }
+    end
+
+    def build_addresses
+      sales_data
+        .uniq { |sale| sale.producer_city && sale.producer_postcode }
+        .map { |sale| build_address(sale) }
+    end
+
+    def build_producers
+      sales_data
+        .uniq(&:producer_id)
+        .map { |sale| build_producer(sale) }
+    end
+
+    def build_supplied_products
+      sales_data
+        .uniq(&:producer_id)
+        .map { |sale| build_supplied_product(sale) }
+    end
+
+    def build_catalogue_items
+      sales_data
+        .uniq(&:producer_id)
+        .map { |sale| build_catalogue_item(sale) }
+    end
+
+    def build_offers
+      sales_data
+        .uniq(&:producer_id)
+        .map { |sale| build_offer(sale) }
+    end
+
+    def build_order_lines
+      sales_data.map { |sale| build_order_line(sale) }
+    end
+
+    def build_orders
+      sales_data.map { |sale| build_order(sale) }
+    end
+
+    # def build_sale_sessions
+    #   sales_data.map { |sale| build_sale_session(sale) }
+    # end
+
+    private
+
+    def build_enterprise(id)
+      DataFoodConsortium::Connector::Enterprise.new(urls.enterprise_url(id))
+    end
+
+    def build_address(sale)
+      DataFoodConsortium::Connector::Address.new(
+        urls.address_url(sale.producer_id),
+        postalCode: sale.producer_postcode,
+        country: nil,
+        latitude: nil,
+        longitude: nil,
+        region: nil,
+        street: nil,
+        city: sale.producer_city
+      )
+    end
+
+    def build_producer(sale)
+      DataFoodConsortium::Connector::Enterprise.new(
+        urls.enterprise_url(sale.producer_id),
```

Comment:
I'm sorry Maikel, what do you mean by identifiable?

Reply:
You are right @mkllnk. The link with the `producer_id` is identifying the enterprise. You can take the url and make another api call that gets all the enterprise data. This could happen with other data as well. @RaggedStaff should we have a placeholder for the object `@id` so that these links don't point to existing endpoints/real data?

Reply:
Yes. That seems like a prudent way to go. 

Can we mask the Enterprise Id in a way that is consistent (maybe a hash) so we can still use it as a reference ?